app/src/main/java/com/github/jorgecastillo/kotlinandroid/data/HeroesRepository.kt

```diff
@@ -2,7 +2,11 @@ package com.github.jorgecastillo.kotlinandroid.data
 
 import com.github.jorgecastillo.kotlinandroid.data.datasource.remote.fetchAllHeroes
 import com.github.jorgecastillo.kotlinandroid.data.datasource.remote.fetchHeroesFromAvengerComics
+import com.karumi.marvelapiclient.model.CharacterDto
+import kategory.HK
 
-fun getHeroesWithCachePolicy() = fetchAllHeroes()
+inline fun <reified F> getHeroesWithCachePolicy(): HK<F, List<CharacterDto>> =
```

Comment:
Sure, I'm not talking about using `AsyncResult`and not `MonadControl` to avoid the reify or the inline, but about the actual need for a user to reify a type with no clear rationale looking at the needs of the function itself. It's a need from the framework here (`kategory`) and I fear the rationale behind it is not that clear for an end user. Not sure if we can fix this in any way inside kategory atm, but that might become confusing in terms of usage.  Just a problem I found when reading this PR ¯\_(ツ)_/¯

But as I see it's completely necessary given the `kategory` implementation we have now.